pyxem/generators/red_intensity_generator.py

```diff
@@ -0,0 +1,201 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Reduced intensity generator and associated tools.
+
+
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+from hyperspy.signals import Signal1D
+
+from pyxem.signals.electron_diffraction1d import ElectronDiffraction1D
+from pyxem.signals.reduced_intensity1d import ReducedIntensity1D
+
+from pyxem.components.scattering_fit_component import ScatteringFitComponent
+from pyxem.utils.ri_utils import scattering_to_signal
+from pyxem.signals import transfer_navigation_axes
+from pyxem.signals import transfer_signal_axes
+
+
+class ReducedIntensityGenerator():
+    """Generates a reduced intensity 1D profile for a specified diffraction radial
+    profile.
+
+
+    Parameters
+    ----------
+    signal : ElectronDiffraction1D
+        An electron diffraction radial profile.
+    """
+
+    def __init__(self, signal, *args, **kwargs):
+        self.signal = signal
+        self.cutoff = [0, signal.axes_manager.signal_axes[0].size - 1]
+        self.nav_size = [signal.axes_manager.navigation_axes[0].size,
+                         signal.axes_manager.navigation_axes[1].size]
+        self.sig_size = [signal.axes_manager.signal_axes[0].size]
+        self.background_fit = None  # added in one of the fits below.
+        self.normalisation = None
+
+    def specify_scattering_calibration(self, calibration):
```

Comment:
Naming here should follow the syntax used in ElectronDiffraction1D - it's probably set_diffraction_calibration() or something.

Reply:
Updated.